File: 20221020_Thursday/20221020.rb
# Leetcode: 12. Integer to Roman.
# https://leetcode.com/problems/integer-to-roman/
# @param {Integer} num
# @return {String}
# Solving Another Task.
def int_to_roman(num)
  answer = 0
  skip = false
  num.split.each_with_index do |char,i|
    next if skip 
    if ?I == char
      if i < num.size - 1 && ?V == num[i+1]
        answer += 4
        skip = true
        next
      end
      if i < num.size - 1 && ?X == num[i+1]
        answer += 9
        skip = true
        next
      end
      answer += 1
    end
    if ?V == char
      answer += 5
    end
    if ?X == char
      if i < num.size - 1 && ?L == num[i+1]
        answer += 40
        skip = true
        next
      end
      if i < num.size - 1 && ?C == num[i+1]
        answer += 90
        skip = true
        next
      end
      answer += 10
    end
    if ?L == char
      answer += 50
    end
    if ?C == char
      if i < num.size - 1 && ?D == num[i+1]
        answer += 400
        skip = true
        next
      end
      if i < num.size - 1 && ?M == num[i+1]
        answer += 900
        skip = true
        next
      end
      answer += 100
    end
    if ?D == char
      answer += 500
    end
    if ?M == char
      answer += 1000
    end
    skip = false
  end
  return answer
end

File: 20221020_Thursday/20221020.go
// Leetcode: 12. Integer to Roman.
// https://leetcode.com/problems/integer-to-roman/
// = = = = = = = = = = = = = =
// Accpeted.
// Thanks God, Jesus Christ!
// = = = = = = = = = = = = = =
// Runtime: 12 ms, faster than 72.72% of Go online submissions for Integer to Roman.
// Memory Usage: 3.2 MB, less than 70.40% of Go online submissions for Integer to Roman.
// 2022.10.20 Daily Challange.
type Val_str struct{
  Val int
  Str string
}
func intToRoman(num int) string {
  answer := ""
  values := []Val_str{
    Val_str{1000, "M"},
    Val_str{900, "CM"},
    Val_str{500, "D"},
    Val_str{400, "CD"},
    Val_str{100, "C"},
    Val_str{90, "XC"},
    Val_str{50, "L"},
    Val_str{40, "XL"},
    Val_str{10, "X"},
    Val_str{9, "IX"},
    Val_str{5, "V"},
    Val_str{4, "IV"},
    Val_str{1,"I"},
  }
  for ; num > 0; {
    for _,pair := range values {
      if pair.Val <= num {
        num -= pair.Val
        answer += pair.Str
        break
      } else {
        continue
      }
    }
  }
  return answer
}
